test/functional/sqlsrv/MsHelper.inc

```diff
@@ -0,0 +1,812 @@
+ * @return string a complete insert sql string
+ */
+function getInsertSqlComplete($tbname, $inputs)
+{
+    $colStr = "INSERT INTO $tbname (";
+    $valStr = "VALUES (";
+    if (empty($inputs)) {
+        echo "getInsertSqlComplete: inputs for inserting a row cannot be empty.\n";
+        return;
+    }
+    foreach ($inputs as $key => $value) {
+        $colStr .= $key . ", ";
+        if (is_null($value)) {
+            echo "getInsertSqlComplete: value provided for input $value is null.\n";
+        } elseif (is_string($value)) {
+            $valStr .= "'" . $value . "', ";
+        } else {
+            $valStr .= $value . ", ";
+        }
+    }
+    $colStr = rtrim($colStr, ", ") . ") ";
+    $valStr = rtrim($valStr, ", ") . ") ";
+    $insertSql = $colStr . $valStr;
+    return $insertSql;
+}
+
+/**
+ * @param  string  $tbname : name of the table for an insert sql
+ * @param  array  $inputs : associative array containing a key value pair of column name and data to put into an insert sql string
+ * @return string an insert sql string with "?" placeholders for all values
+ */
+function getInsertSqlPlaceholders($tbname, $inputs)
+{
+    $colStr = "INSERT INTO $tbname (";
+    $valStr = "VALUES (";
+    if (empty($inputs)) {
+        echo "getInsertSqlPlaceholders: inputs for inserting a row cannot be empty.\n";
+        return;
+    }
+    foreach ($inputs as $key => $value) {
+        $colStr .= $key . ", ";
+    }
+    $colStr = rtrim($colStr, ", ") . ") ";
+    $valStr .= getSeqPlaceholders(count($inputs)) . ") ";
+    $insertSql = $colStr . $valStr;
+    return $insertSql;
+}
+
+/**
+ * @param  string  $spname : name of the stored procedure
+ * @param  int  $num : number of parameters needed for the stored procedure
+ * @return string a call stored procedure sql string with "?" placeholders for all parameters
+ */
+function getCallProcSqlPlaceholders($spname, $num)
+{
+    $callStr = "{CALL $spname (";
+    $callStr .= getSeqPlaceholders($num) . ")} ";
+    return $callStr;
+}
+
+/**
+ * @param  int  $num : number of placeholders needed
+ * @return string a string containing $num number of repeated "?" placeholders delimited by ", "
+ */
+function getSeqPlaceholders($num)
+{
+    if ($num <= 0) {
+        echo "getSeqPlaceholders: num provided for creating a sequence of placeholders cannot be less than 0.\n";
+        return;
+    }
+    $placeholderStr = str_repeat("?, ", $num);
+    $placeholderStr = rtrim($placeholderStr, ", ");
+    return $placeholderStr;
+}
+
+/**
+ * @return bool false if $keystore specified in MsSetup.inc is none or data not encrypted, 
+ *              otherwise return true
+ */
+function isColEncrypted()
+{
+    global $keystore, $dataEncrypted;
+    if ($keystore == KEYSTORE_NONE || !$dataEncrypted) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+/**
+ * @return bool false if ODBC version is less than 17 or SQL Server older than 2016
+ */
+function isQualified($conn)
```

Comment:
I think this should be renamed back to isAEQualified since ODBC17 and SQL Server 13 is only required for AE related features.

Reply:
The method is in the namespace AE so I removed the redundant `AE` from its name